README.md

```diff
@@ -1 +1,116 @@
-# check-commit-hook
\ No newline at end of file
+# check-commit-hook
+
+Based on [1] this repository contains a set of BW pre-commit hooks to be run on every git commit, the same hooks are used by the CI/CD pipeline to check the files.
+
+# Hooks
+
+## Checkpatch
+
+Uses Linux kernel checkpatch to check for coding style errors in C code, Makefiles, etc. It uses a custom configuration file to ignore some errors and enforce others.
+
+### Custom configuration file
+
+The script uses a default configuration file located in `bw_check_commit_hook/data/checkpatch.yaml` but it can be overriden by a custom configuration file and passed to the script using the `--config-file` option.
+
+```
+checkpatch --config-file <config_file> <file>
+```
+
+The configuration file is a YAML file with the following structure:
+
+```yaml
+DIR_CONFIGS:
+  <directory>:
+    errors_enabled:
+      - <checkpatch option>
+      - <checkpatch option>
+      - ...
+    errors_ignored:
+      - <checkpatch option>
+      - <checkpatch option>
+    max_line_length: <number>
+```
+
+#### Magic keys
+
+The configuration file supports some magic keys that make it easier to configure the script.
+
+##### `all`
+
+The `all` key can be used to apply the same configuration to all the files in the repository.
+
+```yaml
+all:
+  errors_enabled:
+    - <checkpatch option>
+    - <checkpatch option>
+    - ...
+  errors_ignored:
+    - <checkpatch option>
+    - <checkpatch option>
+  max_line_length: <number>
+```
+
+##### `ERRORS_COMMON`
+
+The `ERRORS_ENABLED` key can be used as a placeholder for all the errors enabled for all the files, and this needs to be part of `errors_enabled` list to be used.
+
+```yaml
+ERRORS_ENABLED:
+  - <checkpatch option>
+  - <checkpatch option>
+  - ...
+
+DIR_CONFIGS:
+  <directory>:
+    errors_enabled:
+      - ERRORS_ENABLED
+      - <checkpatch option>
+      - <checkpatch option>
+      - ...
+    errors_ignored:
+      - <checkpatch option>
+      - <checkpatch option>
+    max_line_length: <number>
+```
+
+##### `IGNORES_COMMON`
+
+The `IGNORES_COMMON` key can be used as a placeholder for all the errors ignored for all the files, and this needs to be part of `errors_ignored` list to be used.
+
+```yaml
+IGNORES_COMMON:
+  - <checkpatch option>
+  - <checkpatch option>
+  - ...
+
+DIR_CONFIGS:
+  <directory>:
+    errors_enabled:
+      - <checkpatch option>
+      - <checkpatch option>
+      - ...
+    errors_ignored:
+      - IGNORES_COMMON
+      - <checkpatch option>
+      - <checkpatch option>
+    max_line_length: <number>
+```
+
+# Usage
+
+## Pre-commit
+
+To use the hooks in your local repository you need to install pre-commit [1] and add the following to your `.pre-commit-config.yaml` file:
+
+```yaml
+  - repo: https://github.com/bluwireless/check-commit-hook
+    rev: ccd9108
```

Comment:
This would normally be a tag - better to put a generic placeholder rather than a specific commit SHA.

Reply:
Sure, pre-commit shows a warning that non-specific revisions are only updated once, so, I was using the commit SHA, but yes, this should be replaced by a tag once the PR is merged.

Reply:
Sorry, didn't realize that this is a doc :), replaced with xxx.